win32_deps_build.sh

```diff
@@ -83,7 +83,7 @@ case "$OS" in
     ubuntu)
         sudo apt-get update
         sudo env DEBIAN_FRONTEND=noninteractive apt-get -y install \
-            mingw-w64 g++ cmake pkg-config \
+            mingw-w64 g++-12 cmake pkg-config \
```

Comment:
probably should install `g++-mingw-w64` instead ?  but i think we need to upgrade the jenkins work node to ubuntu/noble  first. see https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=g%2B%2B-mingw-w64&searchon=names

Reply:
Hi, we're building the Windows binaries using mingw-llvm: 

https://github.com/ceph/ceph/blob/81a258352afcf376d4237dcd9dab6bb54d5fd6cc/win32_deps_build.sh#L43C15-L44

https://github.com/ceph/ceph/blob/81a258352afcf376d4237dcd9dab6bb54d5fd6cc/mingw_conf.sh#L36-L53

We're using just a few tools from the ``mingw-w64`` package, including dlltool if I remember correctly.

For what is worth, we had lots of problems with mingw-gcc, expecially around winpthreads, which is why we had to switch to mingw-llvm: https://github.com/ceph/ceph/commit/3ce1e4a5c64bc0d71d3e6ad9ece040e4884d0378

Reply:
About the Windows CI job: it builds both Windows and Linux binaries and it seems that the Windows build actually passes:

https://jenkins.ceph.com/job/ceph-windows-pull-requests/44528/consoleFull

```
  WIN32 files zipped to: /home/ubuntu/ceph.zip


real	42m33.561s
user	0m0.305s
sys	0m0.476s
```

It's the Linux build that fails: https://github.com/ceph/ceph-build/blob/e8d18c7aef15fa0424d096ba73a6e0d701a5215a/scripts/ceph-windows/setup_ceph_vstart#L31-L41

Reply:
Since everything is running in a VM, I think all we have to do is to switch the Ubuntu image to 24.04: https://github.com/ceph/ceph-build/blob/e8d18c7aef15fa0424d096ba73a6e0d701a5215a/scripts/ceph-windows/setup_libvirt_ubuntu_vm#L12